### PR TITLE
Methods and tests for jep334 String

### PR DESCRIPTION
**Pull request by theresa-m**
- Java12andUp test framework
- String.describeConstable & tests
- String.resolveConstantDesc & tests

Depends on (see https://github.com/eclipse/openj9/issues/4195): 
- Fix Java 12 build failures: https://github.com/eclipse/openj9/pull/3980
- Constants API stubs: https://github.com/eclipse/openj9/pull/4030
- Java12andUp test setup: https://github.com/eclipse/openj9/pull/4104

Signed-off-by: Theresa Mammarella <Theresa.T.Mammarella@ibm.com>

### CODE REVIEW

**Comment by pshipton** [comment]
String implements Constable, ConstantDesc
http://cr.openjdk.java.net/~iris/se/12/build/latest/api/java.base/java/lang/String.html

**Comment by theresa-m** [comment]
implements Constable, ConstantDesc is included here https://github.com/eclipse/openj9/pull/4030

File: test/functional/Java12andUp/src/org/openj9/test/java_lang/Test_String.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+	public static Logger logger = Logger.getLogger(Test_String.class);
+
+	/*
+	 * Test Java 12 API String.describeConstable()
+	 */
+	@Test(groups = { "level.sanity" })
+	public void testStringDescribeConstable() {
+		String test = "test";
+
+		Optional<String> optionalTest = test.describeConstable();
```
Comment (DanHeidinga) [inline test/functional/Java12andUp/src/org/openj9/test/java_lang/Test_String.java]: use `.orElseThrow()`.  It simplifies the code and lets the test framework handle the unexpected exception

File: test/functional/Java12andUp/src/org/openj9/test/java_lang/Test_String.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+		if (optionalTest.isPresent()) {
+			String describedTestString = optionalTest.get();
+			Assert.assertTrue(test.equals(describedTestString));
```
Comment (DanHeidinga) [inline test/functional/Java12andUp/src/org/openj9/test/java_lang/Test_String.java]: should this test multiple strings?  Are there any corner cases that should be validated?  empty string?

File: test/functional/Java12andUp/src/org/openj9/test/java_lang/Test_String.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+	/*
+	 * Test Java 12 API String.resolveConstantDesc()
+	 */
+	@Test(groups = { "level.sanity" })
+	public void testStringResolveConstantDesc() {
+		String test = "test";
+		MethodHandles.Lookup lookup = MethodHandles.publicLookup();
+
+		String resolvedTest = test.resolveConstantDesc(lookup);
+		Assert.assertTrue(test.equals(resolvedTest));
```
Comment (DanHeidinga) [inline test/functional/Java12andUp/src/org/openj9/test/java_lang/Test_String.java]: This should test with a null lookup as well

**Comment by theresa-m** [comment]
I've updated the tests based on the review from @DanHeidinga 

I've also added the commit from https://github.com/eclipse/openj9/pull/4104 to this pr because adding an empty testng entry does not compile.

**Comment by DanHeidinga** [comment]
Jenkins test sanity zlinux jdk11

**Comment by smlambert** [comment]
Jenkins test sanity zlinux jdk11

**Comment by smlambert** [comment]
previous PR test failed with machine issue (remote call to node failed), retrying to see if transient

**Comment by DanHeidinga** [comment]
Jenkins test sanity zlinux jdk12

**Comment by DanHeidinga** [comment]
@theresa-m Can you rebase this PR so it'll pick up the fix for the jvm interface version # issue?

**Comment by theresa-m** [comment]
yes. done

**Comment by DanHeidinga** [comment]
Jenkins test sanity xlinux jdk12

**Comment by DanHeidinga** [comment]
Jenkins test sanity xlinux jdk12

**Comment by theresa-m** [comment]
failing tests are related to https://github.com/eclipse/openj9/issues/4661

**Comment by DanHeidinga** [comment]
Build failures are in the `com.ibm.j9.jsr292.api.MethodHandleAPI_defineClass.test_defineClass_ClassBytes_CorruptedArray` test and unrelated to this PR
